internal/http/url_handler.go

```diff
@@ -0,0 +1,62 @@
+package http
+
+import (
+	"github.com/Kenny201/go-yandex-shortener.git/internal/domain/shortener/aggregate"
+	"github.com/go-chi/chi/v5"
+	"io"
+	"net/http"
+)
+
+type ShortenerService interface {
+	Put(url string) (string, error)
+	Get(url string) (*aggregate.URL, error)
+}
+
+type ShortenerHandler struct {
+	shortenerService ShortenerService
+}
+
+func NewShortenerHandler(ss ShortenerService) ShortenerHandler {
+	return ShortenerHandler{
+		shortenerService: ss,
+	}
+}
+
+func (sh ShortenerHandler) GetByIDHandler(w http.ResponseWriter, r *http.Request) {
+	id := chi.URLParam(r, "id")
+	url, err := sh.shortenerService.Get(id)
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusNotFound)
+
+		return
+	}
+
+	w.Header().Set("Location", url.OriginalURL().ToString())
+	w.WriteHeader(http.StatusTemporaryRedirect)
+}
+
+func (sh ShortenerHandler) PostHandler(w http.ResponseWriter, r *http.Request) {
+	body, err := io.ReadAll(r.Body)
```

Comment:
Не стоит забывать закрывать тело запроса после чтения.

```suggestion
	body, err := io.ReadAll(r.Body)
	defer r.Body.Close()
```